db/python/tables/project.py

```diff
@@ -50,362 +28,309 @@ class ProjectPermissionsTable:
 
-        secret_manager = self._get_secret_manager_client()
-        secret_path = secret_manager.secret_path(project_id, secret_name)
+    async def _get_project_id_map(self):
+        """
+        Internally cached get_project_id_map
+        """
+        return {p.id: p for p in await self._get_project_rows_internal()}
 
-        response = secret_manager.access_secret_version(
-            request={'name': f'{secret_path}/versions/latest'}
-        )
+    async def _get_project_name_map(self) -> Dict[str, int]:
+        """Get {project_name: project_id} map"""
+        return {p.name: p.id for p in await self._get_project_rows_internal()}
+
+    async def _get_project_by_id(self, project_id: ProjectId) -> Project:
+        """Get project by id"""
+        pmap = await self._get_project_id_map()
+        if project_id not in pmap:
+            raise NotFoundError(f'Could not find project {project_id}')
+        return pmap[project_id]
+
+    async def _get_project_by_name(self, project_name: str) -> Project:
+        """Get project by name"""
+        pmap = await self._get_project_name_map()
+        if project_name not in pmap:
+            raise NotFoundError(f'Could not find project {project_name}')
+        return await self._get_project_by_id(pmap[project_name])
+
+    async def _get_projects_by_ids(
+        self, project_ids: Iterable[ProjectId]
+    ) -> List[Project]:
+        """Get projects by ids"""
+        pids = set(project_ids)
+        pmap = await self._get_project_id_map()
+        missing_pids = pids - set(pmap.keys())
+        if missing_pids:
+            raise NotFoundError(f'Could not find projects {missing_pids}')
+        return [pmap[pid] for pid in pids]
 
-        return response.payload.data.decode('UTF-8')
+    # endregion UNPROTECTED_GETS
 
-    async def check_access_to_project_ids(
-        self,
-        user: str,
-        project_ids: Iterable[ProjectId],
-        readonly: bool,
-        raise_exception=True,
-    ) -> bool:
-        """Check user has access to list of project_ids"""
-        if not project_ids:
-            raise Forbidden(
-                "You don't have access to this resources, as the resource you requested didn't belong to a project"
-            )
-        if self.allow_full_access:
-            return True
-        spids = list(set(project_ids))
-        # do this all at once to save time
-        promises = [
-            self.check_access_to_project_id(
-                user, project_id, readonly=readonly, raise_exception=False
-            )
-            for project_id in spids
-        ]
-        has_access_map = await asyncio.gather(*promises)
+    # region AUTH
 
-        missing_project_ids = [
-            project_id
-            for project_id, has_access in zip(spids, has_access_map)
-            if not has_access
-        ]
+    async def get_all_projects(self, author: str):
+        """Get all projects"""
+        await self.check_project_creator_permissions(author)
```

Comment:
Just a silly question.
Does this mean if user who is not in 'project-creators' and would call api route project/all, would get Forbidden exception?
Should user gets the list of all projects he/she is allowed to view?

Reply:
Great question! There's a /my equivalent route for that, this was handy for specific applications that need all projects or fail